### PR TITLE
[feature]: lambda vpc_config and memory_size

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
Allow lambdas to configure vpc_config and memory_size

### CODE REVIEW

File: aws-lambda-function/outputs.tf [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
   value = aws_cloudwatch_log_group.log.name
 }
 
-output role_name {
+output "role_name" {
   value = aws_iam_role.role.name
 }
 
-output role_id {
+output "role_id" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:26]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`role_id` output has no description

File: aws-lambda-function/outputs.tf [hunk 2]
```diff
@@ -1,28 +1,28 @@
-output arn {
+output "arn" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:1]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`arn` output has no description

File: aws-lambda-function/outputs.tf [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
 }
 
-output qualified_arn {
+output "qualified_arn" {
   description = "If the lambda function is published, the qualified_arn points at the latest version number."
   value       = aws_lambda_function.lambda.qualified_arn
 }
 
-output invoke_arn {
+output "invoke_arn" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:10]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`invoke_arn` output has no description

File: aws-lambda-function/outputs.tf [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
   value = aws_lambda_function.lambda.invoke_arn
 }
 
-output function_name {
+output "function_name" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:14]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`function_name` output has no description

File: aws-lambda-function/outputs.tf [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
   value = aws_lambda_function.lambda.function_name
 }
 
-output log_group_name {
+output "log_group_name" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:18]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`log_group_name` output has no description

File: aws-lambda-function/outputs.tf [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
   value = aws_cloudwatch_log_group.log.name
 }
 
-output role_name {
+output "role_name" {
```
Comment (cziie) [inline aws-lambda-function/outputs.tf:22]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`role_name` output has no description

File: aws-lambda-function/variables.tf [hunk 7]
```diff
@@ -1,110 +1,126 @@
...
 }
 
-variable filename {
+variable "filename" {
   type    = string
   default = null
 }
 
-variable log_retention_in_days {
+variable "log_retention_in_days" {
```
Comment (cziie) [inline aws-lambda-function/variables.tf:71]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`log_retention_in_days` variable has no description

File: aws-lambda-function/variables.tf [hunk 8]
```diff
@@ -1,110 +1,126 @@
...
 }
 
-variable source_code_hash {
+variable "source_code_hash" {
   type    = string
   default = null
 }
 
-variable filename {
+variable "filename" {
```
Comment (cziie) [inline aws-lambda-function/variables.tf:66]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`filename` variable has no description

File: aws-lambda-function/variables.tf [hunk 9]
```diff
@@ -1,110 +1,126 @@
...
 
-variable kms_key_arn {
+variable "kms_key_arn" {
   type        = string
   description = "KMS key used to encrypt environment variables."
   default     = null
 }
 
-variable source_code_hash {
+variable "source_code_hash" {
```
Comment (cziie) [inline aws-lambda-function/variables.tf:61]: 📝 **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>`source_code_hash` variable has no description

File: aws-lambda-function/variables.tf [hunk 10]
```diff
@@ -1,110 +1,126 @@
...
   type        = bool
   description = "Is this lambda going to be used with a Cloufront distribution? If you set this, you will not have control over log retention, and you cannot include environment variables."
   default     = false
 }
 
-variable reserved_concurrent_executions {
+variable "reserved_concurrent_executions" {
   type        = number
   description = "Set reserved_concurrent_executions for this function. See [docs](https://docs.aws.amazon.com/lambda/latest/dg/configuration-concurrency.html)."
   default     = -1 // aws default
```
Comment (cziie) [inline aws-lambda-function/variables.tf:109]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Single line comments should begin with #

File: aws-lambda-function/main.tf [hunk 11]
```diff
@@ -98,15 +109,35 @@ data aws_iam_policy_document lambda_logging_policy {
...
 }
 
-resource aws_iam_role_policy_attachment lambda_logs {
+resource "aws_iam_role_policy_attachment" "lambda_logs" {
   role       = aws_iam_role.role.name
   policy_arn = aws_iam_policy.lambda_logging.arn
 }
+
+// Execution role basic permissions
+data "aws_iam_policy_document" "role" {
```
Comment (cziie) [inline aws-lambda-function/main.tf:126]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>data "aws_iam_policy_document" "role" is declared but not used

File: aws-lambda-function/main.tf [hunk 12]
```diff
@@ -98,15 +109,35 @@ data aws_iam_policy_document lambda_logging_policy {
...
   policy = data.aws_iam_policy_document.lambda_logging_policy.json
 }
 
-resource aws_iam_role_policy_attachment lambda_logs {
+resource "aws_iam_role_policy_attachment" "lambda_logs" {
   role       = aws_iam_role.role.name
   policy_arn = aws_iam_policy.lambda_logging.arn
 }
+
+// Execution role basic permissions
```
Comment (cziie) [inline aws-lambda-function/main.tf:125]: ⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Single line comments should begin with #

**Review by kuannie1 — APPROVED** [review]

**Pull request merged**